docs: improve docs for `withComponentInputBinding` (#61689)
angular/angular: adev/src/content/guide/routing/common-router-tasks.md

</docs-step>

<docs-step title="Add an `input` to the component">

Update the component to have an `input` property matching the name of the parameter.

```ts
id = input<string|undefined>();
effect(() => {
  const heroId = this.id();
  if(heroId) {
    hero = this.service.getHero(heroId)
  }
});
```

</docs-step>
<docs-step title="Optional: Use a default value">
Binded inputs can be set to `undefined` when no route elements match. 

> In this situation, the router will overwrite the default value of the signal input.<br>

Reviewer: I would recommend rewording this. It is not overwriting a default any more than any value “overwrites” a default. The router assigns undefined when no route information matches the key. Same as assigning undefined in a template. The default value applies when no value ever gets assigned to the input. Component input binding _always_ binds to _all_ inputs.

Author: Makes perfect sense, I'll update that.

> It is recommended that such input reflect that by including `undefined` in the typing.

Reviewer: There are many situations where you bind to information that will always be present, such as data or path params. This likely accounts for a large portion of input binding use.

hero = computed(() => this.service.getHero(heroId));
```

</docs-step>
<docs-step title="Optional: Use a default value">
When using input bindings, the router assigns values to all inputs based on the current route.

> In some cases, such as when an optional query parameter is missing, the router will assign `undefined` if no route data matches the input key.

Reviewer: ```suggestion
The router assigns `undefined` if no route data matches the input key, such as when an optional query parameter is missing.
```
Active voice

> If there's a possibility that an input might not be matched by the route, it's recommended to include `undefined` in the `input`'s type.

Reviewer: ```suggestion
You should include `undefined` in the `input`'s type when there's a possibility that an input might not be matched by the route.
```

> In those scenarios, you can provide a default value by either using the `transform` option on the input or managing a local state with a `linkedSignal`.

Reviewer: ```suggestion
Provide a default value by either using the `transform` option on the input or managing a local state with a `linkedSignal`.
```
